src/k8s-engine/minikube.js

```diff
@@ -317,6 +358,38 @@ class Minikube extends EventEmitter {
     });
   }
 
+  async reset(version) {
+    while (this.#currentType !== undefined) {
+      await sleep(500);
+    }
+    if (![K8s.State.STARTED, K8s.State.READY].includes(this.state)) {
+      return;
+    }
+    this.#currentType = 'reset';
+    try {
+      const sudo = ['ssh', '--', 'sudo'];
+
+      await this.exec(...sudo, 'systemctl', 'stop', 'kubelet.service', { stdio: 'inherit' });
+      await this.exec(...sudo, 'rm', '-rf', '/var/lib/k3s/server/db', { stdio: 'inherit' });
+      if (version) {
+        // change versions here
+      }
+      await this.exec(...sudo, 'systemctl', 'start', 'kubelet.service', { stdio: 'inherit' });
+      await this.exec(...sudo, 'systemctl', 'is-active', '--wait', 'kubelet.service', { stdio: 'inherit' });
+      this.#state = this.#state;
```

Comment:
Why the `this.#state = this.#state;` line? Should this be changing the state to something else?

Reply:
Changed the state to always be `Started`; also added installation of rancher (homestead currently) as required.